pkg/adapter/adapter.go

```diff
@@ -37,10 +38,11 @@ func LoadGenesisDoc(cfg *config.Config) (*cmtypes.GenesisDoc, error) {
 
 // Adapter is a struct that will contain an ABCI Application, and will implement the go-execution interface
 type Adapter struct {
-	App        servertypes.ABCI
-	Store      store.Store
-	Mempool    mempool.Mempool
-	MempoolIDs *mempoolIDs
+	App          servertypes.ABCI
+	Store        ds.Batching
```

Comment:
personal preference: this is not used somewhere else and can be private

not sure if this is the same idea as @tac0turtle mentioned below but if you would replace batch with a custom state store type, you could encapsulate load/store and avoid the verbose db type. 
This may even enable features like caching in the future